thunx-spring/src/main/java/eu/xenit/contentcloud/thunx/spring/data/rest/AbacRepositoryInvokerAdapter.java

```diff
@@ -139,8 +141,59 @@ private String toAlias(Class<?> subjectType) {
     }
 
     private BooleanExpression idExpr(Object id, PathBuilder entityPath) {
-        Field idField = BeanUtils.findFieldWithAnnotation(domainType, Id.class);
+        Field idField = DomainObjectUtils.getIdField(domainType);
         PathBuilder idPath = entityPath.get(idField.getName(), id.getClass());
         return idPath.eq(Expressions.constant(id));
     }
+
+
+    static class DomainObjectUtils {
+
+        private static final boolean JAVAX_PERSISTENCE_PRESENT = ClassUtils.isPresent(
+                "javax.persistence.Id", DomainObjectUtils.class.getClassLoader());
+
+        static final Field getIdField(Class<?> domainClass) {
+
+            Optional<Field> javaxPersistenceId = JAVAX_PERSISTENCE_PRESENT
+                    ? DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class)
+                    : Optional.empty();
+
+            return javaxPersistenceId
+                    .or(() -> DomainObjectUtils.findFieldWithAnnotation(domainClass, org.springframework.data.annotation.Id.class))
+                    .orElse(null);
+        }
```

Comment:
I'm thinking this might be more readable as a `if () { } else if() { } else { }` structure. Or do you have a strong preference for only one return statement?

Reply:
the "problem" is that you either need to do the call twice or risk a `ClassNotFoundException`:

```
if (JAVAX_PERSISTENCE_PRESENT && DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class).isPresent()) {
   var field =  DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class);
   return field.get();
}
```

Reply:
```
if (JAVAX_PERSISTENCE_PRESENT) {
   var jpaIdField = DomainObjectUtils.findFieldWithAnnotation(domainClass, javax.persistence.Id.class);
   if (jpaIdField.isPresent()) {
      return jpaIdField.get();
   }
}
```

Reply:
Reworked as if/else

Reply:
I see what you mean, it's fine as-is then.

Reply:
Already refactored :-)